tokio/src/util/linked_list.rs

```diff
@@ -507,13 +507,15 @@ pub(crate) mod tests {
         }
     }
 
+    #[allow(unused_macros)] // This macro is used inside unsafe block but clippy doesn't see that.
     macro_rules! assert_clean {
         ($e:ident) => {{
             assert!($e.pointers.get_next().is_none());
             assert!($e.pointers.get_prev().is_none());
         }};
     }
```

Comment:
Clippy definitely understands unsafe blocks. The unsafe block is not the reason. The problem is that when fuzz tests are compiled, normal tests will not be compiled. Since the assert is used only in normal tests, you get an unused warning in fuzz tests.

Instead, please add this annotation, which says that the macro should only be compiled when compiling normal tests. The macro will be compiled out for fuzz tests.
```suggestion
    #[cfg(test)]
    macro_rules! assert_clean {
        ($e:ident) => {{
            assert!($e.pointers.get_next().is_none());
            assert!($e.pointers.get_prev().is_none());
        }};
    }
```